pkg/types/aws/platform.go

```diff
@@ -19,4 +19,21 @@ type Platform struct {
 	// platform configuration.
 	// +optional
 	DefaultMachinePlatform *MachinePool `json:"defaultMachinePlatform,omitempty"`
+
+	// CustomRegionOverride list contains custom endpoints which will overridde default
+	// service endpoint of AWS Services.
+	// +optional
+	CustomRegionOverride []CustomEndpoint `json:"customRegionOverride,omitempty"`
```

Comment:
if we have decided we're not going to allow users to specify a region for their custom endpoint, then I still feel we could greatly simplify this code by unmarshaling directly into a map[string]string`. 

Terraform only accepts the map[string]string. If further fields were needed for the cloud provider config then it could be hand edited in the manifest.